### PR TITLE
Update coding style guide

### PR DESCRIPTION
**Pull request by rkgrep**

### CODE REVIEW

File: contributions.md [hunk 1]
```diff
@@ -53,10 +53,36 @@ If you discover a security vulnerability within Laravel, please send an e-mail t
...
+
+Short definition syntax for arrays is prefered.
+
+    $array = ['one', 'two', 'three']; //Good
+    $array = array('one', 'two', 'three'); //Bad
+
+When using class names, `::class` constant should be used in most cases.
+
+    $className = Request::class; //Good
+    $className = \Illuminate\Http\Request::class; //Good too
```
Comment (GrahamCampbell) [inline contributions.md]: no, that one's bad

File: contributions.md [hunk 2]
```diff
@@ -53,10 +53,36 @@ If you discover a security vulnerability within Laravel, please send an e-mail t
...
+    use JohnDoe\ExternalLibrary\OneClass;
+    use JohnDoe\ExternalLibrary\TwoClass;
+    use Illuminate\Database\Eloquent\Model;
+    use JohnDoe\ExternalLibrary\Model as JohnDoeModel;
+
 ### DocBlocks
 
 `@param` tags should **not be aligned** and arguments should be separated by **2 spaces**. 
 
+Class hints should use fully qualified names with leading `\`.
```
Comment (GrahamCampbell) [inline contributions.md]: "and" is better rather than "with", since the `\` is not part of the fully qualified name

File: contributions.md (hunk 1 above)
Comment (GrahamCampbell) [inline contributions.md]: apart from in config files, we always import things before using them

**Comment by rkgrep** [comment]
@GrahamCampbell thanks. Fixed it.

File: contributions.md [hunk 3]
```diff
@@ -53,10 +53,38 @@ If you discover a security vulnerability within Laravel, please send an e-mail t
...
+Short definition syntax for arrays is prefered.
+
+    $array = ['one', 'two', 'three']; //Good
+    $array = array('one', 'two', 'three'); //Bad
+
+When using class names as strings, import these classes and use `::class` keyword.
+
+    use Illuminate\Http\Request;
+    
+    $className = Request::class; //Good
```
Comment (GrahamCampbell) [inline contributions.md]: might be a good idea to follow our comment cs, and leave a space after the `//`

**Comment by taylorotwell** [comment]
This class name stuff isn't always the case.

**Pull request closed without merging**